src/Simulation/Native/src/external/fused.hpp

```diff
@@ -133,16 +144,9 @@ class Fused
         }
 
         // New rules of when to stop fusing
-        Fusion::IndexVector qs      = std::vector<unsigned>(1, q);
-        if (fusedgates.predict(qs, cs) > maxFusedSpan || fusedgates.size() >= maxFusedDepth)  flush(wfn);
-        fusedgates.insert(convertMatrix(mat), qs, cs);
-    }
+        Fusion::IndexVector qs = std::vector<unsigned>(1, q);
```

Comment:
Mix-and-match of custom type aliases and underlying built-in types makes the code harder to understand and implies there might be some non-trivial conversion going on here. I'd suggest to replace it with `Fusion::IndexVector qs(1, q);` or `Fusion::IndexVector qs{q};`.